src/libtiled/maprenderer.cpp

```diff
@@ -82,7 +82,30 @@ MapRenderer::~MapRenderer()
 
 QRectF MapRenderer::boundingRect(const ImageLayer *imageLayer) const
 {
-    return QRectF(QPointF(), imageLayer->image().size());
+    QRectF bounds = QRectF(QPointF(), imageLayer->image().size());
+
+    if (!imageLayer->repeatHorizontally() && !imageLayer->repeatVertically())
+        return bounds;
+
+    QRect mapBounds = QRect(QPoint(), map()->size());
+
+    LayerIterator iterator(map(), Layer::TileLayerType);
+    while (TileLayer *tileLayer = static_cast<TileLayer*>(iterator.next()))
+        mapBounds = mapBounds.united(tileLayer->bounds());
+
+    if (mapBounds.size() == QSize(0, 0))
+        mapBounds.setSize(QSize(1, 1));
```

Comment:
What about calling `mapBoundingRect`?

Reply:
It seems to me that the method is not implemented for maprenderer. It is for others renderers, though.

Reply:
Yes, this is a pure virtual function, but you can still call it. It is not possible to instantiate a `MapRenderer` because of this and other pure virtual functions. All instances will be from one of its subclasses, and the virtual function call will end up at the appropriate implementation (which is why function is virtual).

Reply:
I get what you mean now, sorry. But I think this is irrelevant now.